PR: https://github.com/odoo/odoo/pull/282867

From: eb01adbca9d129b41cbfbf48634eb174ca464c53
From: Arnold Moyaux
Date: 2026-09-10 11:43:56

Breaking data model changes scores: del:15 + add:0, change matches:
-    _inherit = 'account.account'
-    account_stock_variation_id = fields.Many2one(
-    account_stock_variation_active = fields.Boolean(related='account_stock_variation_id.active', string="Variation Account Active")
-    account_stock_expense_id = fields.Many2one(
-    account_stock_expense_active = fields.Boolean(related='account_stock_expense_id.active', string="Expense Account Active")
-    valuation = fields.Selection(
-    property_valuation = fields.Selection(
-    property_stock_journal = fields.Many2one(
-    property_stock_valuation_account_id = fields.Many2one(
-    property_stock_valuation_account_active = fields.Boolean(related='property_stock_valuation_account_id.active', string="Stock Valuation Account Active")
-    account_stock_variation_id = fields.Many2one(
-    account_stock_variation_active = fields.Boolean(related='account_stock_variation_id.active', string="Stock Variation Account Active")
-    account_stock_journal_id = fields.Many2one(
-    account_stock_valuation_id = fields.Many2one(
-    inventory_valuation = fields.Selection(

Total Changes: 120

[REF] account: move inventory valuation account

Part-of: odoo/odoo#282867
Related: odoo/enterprise#128165
Related: odoo/upgrade#11082
Signed-off-by: William Henrotin (whe) <whe@odoo.com>

================================= pseudo patch: =================================

--- a/addons/stock_account/models/__init__.py
+++ b/addons/stock_account/models/__init__.py
@@ -1,8 +1,6 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from . import account_account
-from . import account_chart_template
 from . import account_move
 from . import account_move_line
 from . import analytic_account

--- a/addons/stock_account/models/account_account.py
+++ b/None
@@ -1,14 +0,0 @@
-from odoo import fields, models
-
-
-class AccountAccount(models.Model):
-    _inherit = 'account.account'
-
-    account_stock_variation_id = fields.Many2one(
-        'account.account', string='Variation Account',
-        help="At closing, register the inventory variation of the period into a specific account")
-    account_stock_variation_active = fields.Boolean(related='account_stock_variation_id.active', string="Variation Account Active")
-    account_stock_expense_id = fields.Many2one(
-        'account.account', string='Expense Account',
-        help="Counterpart used at closing for accounting adjustments to inventory valuation.")
-    account_stock_expense_active = fields.Boolean(related='account_stock_expense_id.active', string="Expense Account Active")

--- a/addons/stock_account/models/account_chart_template.py
+++ b/None
@@ -1,29 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import models, _
-from odoo.addons.account.models.chart_template import template
-
-
-class AccountChartTemplate(models.AbstractModel):
-    _inherit = "account.chart.template"
-
-    @template(model='account.journal')
-    def _get_stock_account_journal(self, template_code):
-        return {
-            'inventory_valuation': {
-                'name': _('Inventory Valuation'),
-                'code': 'STJ',
-                'type': 'general',
-                'sequence': 10,
-                'show_on_dashboard': False,
-            },
-        }
-
-    @template(model='res.company')
-    def _get_stock_res_company(self, template_code):
-        return {
-            self.env.company.id: {
-                'account_stock_journal_id': 'inventory_valuation',
-            },
-        }

--- a/addons/stock_account/models/product.py
+++ b/addons/stock_account/models/product.py
@@ -27,32 +27,12 @@ class ProductTemplate(models.Model):
         ],
         compute='_compute_cost_method',
     )
-    valuation = fields.Selection(
-        string="Valuation",
-        selection=[
-            ('periodic', 'Periodic (at closing)'),
-            ('real_time', 'Perpetual (at invoicing)'),
-        ],
-        compute='_compute_valuation', search='_search_valuation',
-    )
     lot_valuated = fields.Boolean(
         string="Valuation by Lot/Serial",
         compute='_compute_lot_valuated', store=True, readonly=False,
         help="If checked, the valuation will be specific by Lot/Serial number.",
     )
 
-    def _search_valuation(self, operator, value):
-        if operator != '=':
-            raise UserError(self.env._("You can only use the '=' operator to search on valuation field."))
-        if value not in ['periodic', 'real_time']:
-            raise UserError(self.env._("Only the value 'periodic' and 'real_time' are accepted to search on valuation field."))
-        domain_categ = Domain([('categ_id.property_valuation', operator, value)])
-        domain_company = Domain(['|', ('categ_id.property_valuation', '=', False), ('categ_id', '=', False), ('company_id.inventory_valuation', operator, value)])
-
-        if self.env.company.inventory_valuation and self.env.company.inventory_valuation == value:
-            domain_company = Domain(['|', ('categ_id.property_valuation', '=', False), ('categ_id', '=', False), '|', ('company_id.inventory_valuation', operator, value), ('company_id', '=', False)])
-        return domain_company | domain_categ
-
     @api.depends('tracking', 'is_storable')
     def _compute_lot_valuated(self):
         for product in self:
@@ -71,15 +51,6 @@ class ProductTemplate(models.Model):
                 or company.sudo().cost_method
             )
 
-    @api.depends_context('company')
-    @api.depends('categ_id.property_valuation')
-    def _compute_valuation(self):
-        for product_template in self:
-            company = product_template.company_id
-            if not company or self.env.company.filtered_domain([('id', 'child_of', company.id)]):
-                company = self.env.company
-            product_template.valuation = product_template.categ_id.with_company(company).property_valuation or company.inventory_valuation
-
     def write(self, vals):
         product_ids_to_update = set()
         lot_ids_to_update = set()
@@ -825,16 +796,6 @@ class ProductCategory(models.Model):
     anglo_saxon_accounting = fields.Boolean(
         string="Use Anglo-Saxon Accounting", compute="_compute_anglo_saxon_accounting",
         help="If checked, the product will be valued using the Anglo-Saxon accounting method.")
-    property_valuation = fields.Selection(
-        string="Inventory Valuation",
-        selection=[
-            ('periodic', 'Periodic (at closing)'),
-            ('real_time', 'Perpetual (at invoicing)'),
-        ],
-        company_dependent=True, copy=True, tracking=True,
-        help="""Periodic: The accounting entries are suggested manually in the inventory valuation report.
-        Perpetual: An accounting entry is automatically created to value the inventory when a product is billed or invoiced.
-        """)
     property_cost_method = fields.Selection(
         string="Costing Method",
         selection=[
@@ -850,23 +811,11 @@ class ProductCategory(models.Model):
         """,
         tracking=True,
     )
-    property_stock_journal = fields.Many2one(
-        'account.journal', 'Stock Journal', company_dependent=True,
-        help="When doing automated inventory valuation, this is the Accounting Journal in which entries will be automatically posted when stock moves are processed.")
-    property_stock_valuation_account_id = fields.Many2one(
-        'account.account', 'Stock Valuation Account', company_dependent=True, ondelete='restrict',
-        check_company=True,
-        help="""When automated inventory valuation is enabled on a product, this account will hold the current value of the products.""")
-    property_stock_valuation_account_active = fields.Boolean(related='property_stock_valuation_account_id.active', string="Stock Valuation Account Active")
     property_price_difference_account_id = fields.Many2one(
         'account.account', 'Price Difference Account', company_dependent=True, ondelete='restrict',
         check_company=True,
         help="""With perpetual valuation, this account will hold the price difference between the standard price and the bill price.""")
     property_price_difference_account_active = fields.Boolean(related='property_price_difference_account_id.active', string="Price Difference Account Active")
-    account_stock_variation_id = fields.Many2one(
-        'account.account', string="Stock Variation Account", readonly=False,
-        related="property_stock_valuation_account_id.account_stock_variation_id")
-    account_stock_variation_active = fields.Boolean(related='account_stock_variation_id.active', string="Stock Variation Account Active")
 
     @api.depends_context('company')
     def _compute_anglo_saxon_accounting(self):

--- a/addons/stock_account/models/res_company.py
+++ b/addons/stock_account/models/res_company.py
@@ -11,20 +11,6 @@ from odoo.addons.base.models.res_company import company_default_for
 class ResCompany(models.Model):
     _inherit = "res.company"
 
-    account_stock_journal_id = fields.Many2one(
-        'account.journal',
-        string='Stock Journal',
-        **company_default_for('account_stock_journal_id', 'product.category', 'property_stock_journal'),
-        check_company=True,
-    )
-
-    account_stock_valuation_id = fields.Many2one(
-        'account.account',
-        string='Stock Valuation Account',
-        **company_default_for('account_stock_valuation_id', 'product.category', 'property_stock_valuation_account_id'),
-        check_company=True,
-    )
-
     account_production_wip_account_id = fields.Many2one('account.account', string='Production WIP Account', check_company=True)
     account_production_wip_overhead_account_id = fields.Many2one('account.account', string='Production WIP Overhead Account', check_company=True)
 
@@ -38,16 +24,6 @@ class ResCompany(models.Model):
         default='manual',
         required=True)
 
-    inventory_valuation = fields.Selection(
-        string='Valuation',
-        selection=[
-            ('periodic', 'Periodic (at closing)'),
-            ('real_time', 'Perpetual (at invoicing)'),
-        ],
-        **company_default_for('inventory_valuation', 'product.category', 'property_valuation'),
-        default='periodic',
-    )
-
     cost_method = fields.Selection(
         string="Cost Method",
         selection=[
